packages/squiggle-lang/__tests__/Distributions/Samples_test.res

```diff
@@ -0,0 +1,46 @@
+open Jest
+open Expect
+
+let makeTest = (~only=false, str, item1, item2) =>
+  only
+    ? Only.test(str, () => expect(item1) -> toEqual(item2))
+    : test(str, () => expect(item1) -> toEqual(item2))
+
+describe("Continuous and discrete splits", () => {
+  makeTest(
+    "check splits one",
+    SampleSet.Internals.T.splitContinuousAndDiscrete([1.432, 1.33455, 2.0]),
+    ([1.432, 1.33455, 2.0], E.FloatFloatMap.empty()),
+  )
+  makeTest(
+    "check splits two",
+    SampleSet.Internals.T.splitContinuousAndDiscrete([
+      1.432,
+      1.33455,
+      2.0,
+      2.0,
+      2.0,
+      2.0,
+    ]) |> (((c, disc)) => (c, disc |> E.FloatFloatMap.toArray)),
+    ([1.432, 1.33455], [(2.0, 4.0)]),
+  )
+
+  let makeDuplicatedArray = count => {
+    let arr = Belt.Array.range(1, count) |> E.A.fmap(float_of_int)
+    let sorted = arr |> Belt.SortArray.stableSortBy(_, compare)
+    E.A.concatMany([sorted, sorted, sorted, sorted]) |> Belt.SortArray.stableSortBy(_, compare)
+  }
+
+  let (_, discrete) = SampleSet.Internals.T.splitContinuousAndDiscrete(
+    makeDuplicatedArray(10),
+  )
+  let toArr = discrete |> E.FloatFloatMap.toArray
+  makeTest("splitMedium at count=10", toArr |> Belt.Array.length, 10)
+
+  let (_c, discrete) = SampleSet.Internals.T.splitContinuousAndDiscrete(
+    makeDuplicatedArray(500),
+  )
+  let toArr = discrete |> E.FloatFloatMap.toArray
+  makeTest("splitMedium at count=500", toArr |> Belt.Array.length, 500)
```

Comment:
This sections (from line 28) seems a bit too messy to me. 

I could see something like, 
```
 makeTest("splitMedium at count=10", duplicatedArray(10) |> split -> Belt.Array.length, 10)
```

Reply:
I also think this section could use some more descriptions to make it clear what's happening / what's being tested

Reply:
You mean more `describe` blocks with less tests per block?

Reply:
Or just move both ``makeTest`` calls to be next to each other, on the bottom. Right now you have to notice that ``toArr`` is changed between them.

I'm used to having tests with blocks that contain all the test-specific code.

Reply:
committing this in in a sec.